src/UserSpaceInstrumentation/CMakeLists.txt

```diff
@@ -56,7 +56,16 @@ target_link_libraries(UserSpaceInstrumentation PUBLIC
 # executed on function entry/exit of an instrumented function.
 add_library(OrbitUserSpaceInstrumentation SHARED)
 
-set_target_properties(OrbitUserSpaceInstrumentation PROPERTIES OUTPUT_NAME "orbituserspaceinstrumentation")
+# The linker flags mark all symbols coming from libraries as private and ensure that they will be hidden in the shared library.
+# The visibility properties ensure that symbols coming from this module's source files are considered private unless
+# explicitly marked visible in the source code file.
+# This avoids functions and types that we don't want to explicitly expose leaking into the target process and avoids
+# them potentially being used outside of this library.
```

Comment:
I would suggest the following, to separate "expose" and "leaking", which belong to different sentences, and because I think "avoid"+"-ing" is used when the second verb also refers to the subject of "avoid":
```suggestion
# This prevents functions and types that we don't want to explicitly expose from leaking into the target process and from
# potentially being used outside of this library.
```

Reply:
Update Text. Thanks, Davide!